naucse/routes.py

```diff
@@ -265,3 +266,24 @@ def session_url(session):
                            homework_section=homework_section,
                            link_section=link_section,
                            cheatsheet_section=cheatsheet_section)
+
+@app.route('/<course:course>/calendar/')
+def course_calendar(course):
+    if not course.start_date:
+        abort(404)
+    months = []
+    year = course.start_date.year
+    month = course.start_date.month
+    while (year, month) <= (course.end_date.year, course.end_date.month):
```

Comment:
I am wondering, if there possibly isn't already a function, that does this?

```pycon
>>> tuple((d.year, d.month) for d in dateutil.rrule.rrule(dateutil.rrule.MONTHLY, dtstart=datetime.date(year=2017, month=1, day=1), until=datetime.date(year=2018, month=2, day=1)))
((2017, 1), (2017, 2), (2017, 3), (2017, 4), (2017, 5), (2017, 6), (2017, 7), (2017, 8), (2017, 9), (2017, 10), (2017, 11), (2017, 12), (2018, 1), (2018, 2))
```

Now I don't know if this would be any better, faster or more readable (probably not). However when dealing with dates, I'd probably prefer to stick with the date type. (I know, those are in fact months...)

Anyway: please **make it a function** (regardless of the implementation). That would make it:

 * testable
 * easily refactored if needed
 * more readable for a reader of this route (if you pick a descriptive name, maybe `months_range(start_date, end_date)`)

Reply:
I think `(year, month)` tuples are a natural representation here. Datetime objects don't offer much functionality for using them as months.
I went with `list_months`; I didn't use “range” as the end date is included.